L1Trigger/TrackFindingTracklet/src/TrackletCalculatorDisplaced.cc

```diff
@@ -1500,14 +1746,203 @@ void TrackletCalculatorDisplaced::exacttracklet(double r1,
+                                                 double phider[4],
+                                                 double zder[4],
+                                                 double phiprojdisk[5],
+                                                 double rprojdisk[5],
+                                                 double phiderdisk[5],
+                                                 double rderdisk[5]) {
+  double a = 1.0 / ((r1 - r2) * (r1 - r3));
+  double b = 1.0 / ((r1 - r2) * (r2 - r3));
+  double c = 1.0 / ((r1 - r3) * (r2 - r3));
+
+  // first iteration in r-phi plane
+  double halfRinv_0 = -phi1 * r1 * a + phi2 * r2 * b - phi3 * r3 * c;
+  // double phi0_0 = -phi1 * r1 * (r2 + r3) * a
+  //                 + phi2 * r2 * (r1 + r3) * b
+  //                 - phi3 * r3 * (r1 + r2) * c;
+  double d0_0 = r1 * r2 * r3 * (-phi1 * a + phi2 * b - phi3 * c);
+
+  // corrections to phi1, phi2, and phi3
+  double r = r2, z = z2;
+  if (take3)
+    r = r3, z = z3;
+
+  double d0OverR1 = d0_0 * rzmeanInv_[0] * (ndisks > 2 ? fabs((z - z1) / (r - r1)) : 1.0);
+  double d0OverR2 = d0_0 * rzmeanInv_[1] * (ndisks > 1 ? fabs((z - z1) / (r - r1)) : 1.0);
+  double d0OverR3 = d0_0 * rzmeanInv_[2] * (ndisks > 0 ? fabs((z - z1) / (r - r1)) : 1.0);
+
+  double d0OverR = d0OverR2;
+  if (take3)
+    d0OverR = d0OverR3;
+
+  double c1 = d0_0 * halfRinv_0 * d0OverR1 + 2.0 * d0_0 * halfRinv_0 * r1 * halfRinv_0 +
+              (1.0 / 6.0) * pow(-r1 * halfRinv_0 - d0OverR1, 3);
+  double c2 = d0_0 * halfRinv_0 * d0OverR2 + 2.0 * d0_0 * halfRinv_0 * r2 * halfRinv_0 +
+              (1.0 / 6.0) * pow(-r2 * halfRinv_0 - d0OverR2, 3);
+  double c3 = d0_0 * halfRinv_0 * d0OverR3 + 2.0 * d0_0 * halfRinv_0 * r3 * halfRinv_0 +
+              (1.0 / 6.0) * pow(-r3 * halfRinv_0 - d0OverR3, 3);
+
+  double phi1c = phi1 - c1;
+  double phi2c = phi2 - c2;
+  double phi3c = phi3 - c3;
+
+  // second iteration in r-phi plane
+  double halfRinv = -phi1c * r1 * a + phi2c * r2 * b - phi3c * r3 * c;
+  phi0 = -phi1c * r1 * (r2 + r3) * a + phi2c * r2 * (r1 + r3) * b - phi3c * r3 * (r1 + r2) * c;
+  d0 = r1 * r2 * r3 * (-phi1c * a + phi2c * b - phi3c * c);
+
+  t = ((z - z1) / (r - r1)) * (1. + d0 * halfRinv - 0.5 * d0OverR1 * d0OverR -
+                               (1. / 6.) * (r1 * r1 + r2 * r2 + r1 * r2) * halfRinv_0 * halfRinv_0);
+  z0 = z1 -
+       t * r1 * (1.0 - d0_0 * halfRinv_0 - 0.5 * d0OverR1 * d0OverR1 + (1.0 / 6.0) * r1 * r1 * halfRinv_0 * halfRinv_0);
+
+  rinv = 2.0 * halfRinv;
+  phi0 += -phimin_;
+
+  for (unsigned int i = 0; i < toR_.size(); i++) {
+    approxproj(halfRinv,
+               phi0,
+               d0,
+               t,
+               z0,
+               halfRinv_0,
+               d0_0,  // added _0 version for high term calculations
+               toR_.at(i),
+               phiproj[i],
+               phider[i],
+               zproj[i],
+               zder[i]);
+    // phiproj[i] -= 0.171;
```

Comment:
@gabhijith I would suggest removing this and similar comments, assuming we no longer have to apply this magic offset.